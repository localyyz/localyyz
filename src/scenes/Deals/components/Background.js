import React from "react";
import { View, StyleSheet, Text } from "react-native";

// third party
import { computed } from "mobx";
import { inject, observer } from "mobx-react/native";
import Video from "react-native-video";
import LinearGradient from "react-native-linear-gradient";
import PropTypes from "prop-types";

// custom
import { Colours, Sizes, Styles } from "localyyz/constants";
import { Timer } from "localyyz/components";
import { clocksVideo, girlVideo, staticVideo } from "localyyz/assets";

@inject(stores => ({
  currentStatus: stores.dealStore.currentStatus,
  currentTimerTargetArray: stores.dealStore.currentTimerTargetArray
}))
@observer
export default class Background extends React.Component {
  static propTypes = {
    currentTimerTargetArray: PropTypes.array.isRequired,
    currentStatus: PropTypes.number.isRequired,
    onComplete: PropTypes.func
  };

  @computed
  get content() {
    return [
      { video: girlVideo, content: DealWaitingContent },
      { video: clocksVideo, content: DealActiveContent },
      { video: girlVideo, content: DealMissedContent },
      { video: staticVideo, content: DealExpiredContent }
    ][this.props.currentStatus];
  }

  render() {
    return (
      <View style={styles.container}>
        <Video
          muted
          repeat
          playWhenInactive
          resizeMode="cover"
          source={this.content.video}
          style={styles.video}/>
        <View style={styles.content}>
          <LinearGradient
            colors={[Colours.MenuBackground, Colours.BlackTransparent]}
            start={{ y: 1, x: 0 }}
            end={{ y: 0, x: 0 }}
            style={styles.gradient}/>
        </View>
        <this.content.content
          {...this.props}
          target={this.props.currentTimerTargetArray}
          onComplete={this.onComplete}/>
      </View>
    );
  }
}

class DealWaitingContent extends React.Component {
  render() {
    return (
      <View style={[styles.content, styles.centered]}>
        <Text style={[styles.title, styles.header]}>Today at noon</Text>
        <Text style={styles.title}>sale starts in</Text>
        <Text style={styles.title}>
          <Timer
            target={this.props.target}
            onComplete={this.props.onComplete}/>
        </Text>
      </View>
    );
  }
}

class DealActiveContent extends React.Component {
  render() {
    return (
      <View style={[styles.content, styles.centered]}>
        <Text style={styles.title}>{"#DOTD is live!"}</Text>
        <Text style={[styles.title, styles.header]}>
          <Timer
            target={this.props.target}
            onComplete={this.props.onComplete}/>
        </Text>
      </View>
    );
  }
}

class DealMissedContent extends React.Component {
  render() {
    return (
      <View style={[styles.content, styles.centered]}>
        <Text style={[styles.title, styles.header]}>You missed it today</Text>
        <Text style={styles.title}>next deal tomorrow at noon</Text>
        <Text style={styles.title}>
          <Timer
            target={this.props.target}
            onComplete={this.props.onComplete}/>
        </Text>
      </View>
    );
  }
}

class DealExpiredContent extends React.Component {
  render() {
    return (
      <View style={[styles.content, styles.centered]}>
        <Text style={[styles.title, styles.header]}>{".. and it's over"}</Text>
        <Text style={[styles.title, styles.compressed]}>
          next deal tomorrow at noon
        </Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {},

  content: {
    ...Styles.Overlay,
    justifyContent: "flex-end"
  },

  centered: {
    alignItems: "center",
    justifyContent: "center"
  },

  video: {
    height: Sizes.Height / 4,
    width: Sizes.Width
  },

  gradient: {
    height: Sizes.Height / 3,
    width: Sizes.Width
  },

  header: {
    ...Styles.Oversized,
    marginVertical: Sizes.InnerFrame
  },

  compressed: {
    marginHorizontal: Sizes.Width / 4
  },

  title: {
    ...Styles.Text,
    ...Styles.Title,
    ...Styles.Alternate,
    textAlign: "center",
    marginHorizontal: Sizes.OuterFrame
  }
});
