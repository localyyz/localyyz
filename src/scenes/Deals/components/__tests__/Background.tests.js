import React from "react";
import Renderer from "react-test-renderer";

// local
import Background from "../Background";
const Component = Background;

// constants
const NAME = "Deals/UpcomingCard";
const PROPS = {
  currentTimerTargetArray: []
};

describe(NAME, () => {
  it(`${NAME}: should render properly - Deal Waiting Content`, () => {
    let props = { ...PROPS, currentStatus: 0 };
    let comp = Renderer.create(<Component.wrappedComponent {...props} />);
    expect(comp.getInstance()).toBeDefined();
    expect(comp.toJSON()).toMatchSnapshot();
  });

  it(`${NAME}: should render properly - Deal Active Content`, () => {
    let props = { ...PROPS, currentStatus: 1 };
    let comp = Renderer.create(<Component.wrappedComponent {...props} />);
    expect(comp.getInstance()).toBeDefined();
    expect(comp.toJSON()).toMatchSnapshot();
  });

  it(`${NAME}: should render properly - Deal Missed Content`, () => {
    let props = { ...PROPS, currentStatus: 2 };
    let comp = Renderer.create(<Component.wrappedComponent {...props} />);
    expect(comp.getInstance()).toBeDefined();
    expect(comp.toJSON()).toMatchSnapshot();
  });

  it(`${NAME}: should render properly - Deal Expired Content`, () => {
    let props = { ...PROPS, currentStatus: 3 };
    let comp = Renderer.create(<Component.wrappedComponent {...props} />);
    expect(comp.getInstance()).toBeDefined();
    expect(comp.toJSON()).toMatchSnapshot();
  });
});
