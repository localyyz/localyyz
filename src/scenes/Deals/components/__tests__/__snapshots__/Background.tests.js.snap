// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deals/UpcomingCard Deals/UpcomingCard: should render properly - Deal Active Content 1`] = `
<View
  style={Object {}}
>
  <Video
    muted={true}
    playWhenInactive={true}
    repeat={true}
    resizeMode="cover"
    source={
      Object {
        "testUri": "../../../src/assets/videos/clocks.mp4",
      }
    }
    style={
      Object {
        "height": 444.6666666666667,
        "width": 750,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "justifyContent": "flex-end",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <BVLinearGradient
      colors={
        Array [
          4278190080,
          0,
        ]
      }
      endPoint={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      locations={null}
      startPoint={
        Object {
          "x": 0,
          "y": 1,
        }
      }
      style={
        Object {
          "height": 667,
          "width": 750,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "bottom": 0,
          "justifyContent": "flex-end",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0)",
          "borderBottomColor": "#FFFFFF",
          "color": "#FFFFFF",
          "fontSize": 22,
          "fontWeight": "700",
          "marginHorizontal": 25,
          "textAlign": "center",
        }
      }
    >
      Today's #DOTD is live!
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0)",
            "borderBottomColor": "#FFFFFF",
            "color": "#FFFFFF",
            "fontSize": 22,
            "fontWeight": "700",
            "marginHorizontal": 25,
            "textAlign": "center",
          },
          Object {
            "fontSize": 32,
            "fontWeight": "700",
            "lineHeight": 38.4,
            "marginVertical": 15,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        target={Array []}
      >
        00:00:00
      </Text>
    </Text>
  </View>
</View>
`;

exports[`Deals/UpcomingCard Deals/UpcomingCard: should render properly - Deal Expired Content 1`] = `
<View
  style={Object {}}
>
  <Video
    muted={true}
    playWhenInactive={true}
    repeat={true}
    resizeMode="cover"
    source={
      Object {
        "testUri": "../../../src/assets/videos/static.mp4",
      }
    }
    style={
      Object {
        "height": 444.6666666666667,
        "width": 750,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "justifyContent": "flex-end",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <BVLinearGradient
      colors={
        Array [
          4278190080,
          0,
        ]
      }
      endPoint={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      locations={null}
      startPoint={
        Object {
          "x": 0,
          "y": 1,
        }
      }
      style={
        Object {
          "height": 667,
          "width": 750,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "bottom": 0,
          "justifyContent": "flex-end",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0)",
            "borderBottomColor": "#FFFFFF",
            "color": "#FFFFFF",
            "fontSize": 22,
            "fontWeight": "700",
            "marginHorizontal": 25,
            "textAlign": "center",
          },
          Object {
            "fontSize": 32,
            "fontWeight": "700",
            "lineHeight": 38.4,
            "marginVertical": 15,
          },
        ]
      }
    >
      .. and it's over
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0)",
            "borderBottomColor": "#FFFFFF",
            "color": "#FFFFFF",
            "fontSize": 22,
            "fontWeight": "700",
            "marginHorizontal": 25,
            "textAlign": "center",
          },
          Object {
            "marginHorizontal": 187.5,
          },
        ]
      }
    >
      next deal tomorrow at noon
    </Text>
  </View>
</View>
`;

exports[`Deals/UpcomingCard Deals/UpcomingCard: should render properly - Deal Missed Content 1`] = `
<View
  style={Object {}}
>
  <Video
    muted={true}
    playWhenInactive={true}
    repeat={true}
    resizeMode="cover"
    source={
      Object {
        "testUri": "../../../src/assets/videos/girl.mp4",
      }
    }
    style={
      Object {
        "height": 444.6666666666667,
        "width": 750,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "justifyContent": "flex-end",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <BVLinearGradient
      colors={
        Array [
          4278190080,
          0,
        ]
      }
      endPoint={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      locations={null}
      startPoint={
        Object {
          "x": 0,
          "y": 1,
        }
      }
      style={
        Object {
          "height": 667,
          "width": 750,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "bottom": 0,
          "justifyContent": "flex-end",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0)",
            "borderBottomColor": "#FFFFFF",
            "color": "#FFFFFF",
            "fontSize": 22,
            "fontWeight": "700",
            "marginHorizontal": 25,
            "textAlign": "center",
          },
          Object {
            "fontSize": 32,
            "fontWeight": "700",
            "lineHeight": 38.4,
            "marginVertical": 15,
          },
        ]
      }
    >
      You missed it today
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0)",
          "borderBottomColor": "#FFFFFF",
          "color": "#FFFFFF",
          "fontSize": 22,
          "fontWeight": "700",
          "marginHorizontal": 25,
          "textAlign": "center",
        }
      }
    >
      next deal tomorrow at noon
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0)",
          "borderBottomColor": "#FFFFFF",
          "color": "#FFFFFF",
          "fontSize": 22,
          "fontWeight": "700",
          "marginHorizontal": 25,
          "textAlign": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        target={Array []}
      >
        00:00:00
      </Text>
    </Text>
  </View>
</View>
`;

exports[`Deals/UpcomingCard Deals/UpcomingCard: should render properly - Deal Waiting Content 1`] = `
<View
  style={Object {}}
>
  <Video
    muted={true}
    playWhenInactive={true}
    repeat={true}
    resizeMode="cover"
    source={
      Object {
        "testUri": "../../../src/assets/videos/girl.mp4",
      }
    }
    style={
      Object {
        "height": 444.6666666666667,
        "width": 750,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "justifyContent": "flex-end",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <BVLinearGradient
      colors={
        Array [
          4278190080,
          0,
        ]
      }
      endPoint={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      locations={null}
      startPoint={
        Object {
          "x": 0,
          "y": 1,
        }
      }
      style={
        Object {
          "height": 667,
          "width": 750,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "bottom": 0,
          "justifyContent": "flex-end",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0)",
            "borderBottomColor": "#FFFFFF",
            "color": "#FFFFFF",
            "fontSize": 22,
            "fontWeight": "700",
            "marginHorizontal": 25,
            "textAlign": "center",
          },
          Object {
            "fontSize": 32,
            "fontWeight": "700",
            "lineHeight": 38.4,
            "marginVertical": 15,
          },
        ]
      }
    >
      Today at noon
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0)",
          "borderBottomColor": "#FFFFFF",
          "color": "#FFFFFF",
          "fontSize": 22,
          "fontWeight": "700",
          "marginHorizontal": 25,
          "textAlign": "center",
        }
      }
    >
      sale starts in
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0)",
          "borderBottomColor": "#FFFFFF",
          "color": "#FFFFFF",
          "fontSize": 22,
          "fontWeight": "700",
          "marginHorizontal": 25,
          "textAlign": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        target={Array []}
      >
        00:00:00
      </Text>
    </Text>
  </View>
</View>
`;
